contracts/contracts/RequestManager.sol

```diff
@@ -92,34 +92,25 @@ contract RequestManager is Ownable {
     mapping (uint256 => Request) public requests;
     mapping (uint256 => Claim) public claims;
 
-    uint256 public gasPrice = 5e9;
-    uint256 private serviceFeePPM = 45_000;  //4.5%
+    uint256 public minLpFee = 5 ether;  // 5e18
+    uint256 public lpFeePPM = 1_000;  // 0.1% of the token amount being transferred
+    uint256 public protocolFeePPM = 0; // 0% of the token amount being transferred
 
-    // beamer fee tracking
-    uint256 public collectedBeamerFees = 0;
+    // Protocol fee tracking: ERC20 token address => amount
+    mapping (address => uint256) public collectedProtocolFees;
 
-    // The optimizer should take care of eval'ing this
-    function gasReimbursementFee() public view returns (uint256) {
-        uint256 fillGas = 67105;
-        uint256 claimGas = 154634;
-        uint256 withdrawGas = 64081;
-
-        return (fillGas + claimGas + withdrawGas) * gasPrice;
-    }
-
-    function lpServiceFee() public view returns (uint256) {
-        return gasReimbursementFee() * serviceFeePPM / 1_000_000;
+    function lpFee(uint256 amount) public view returns (uint256) {
+        return Math.max(minLpFee, amount * lpFeePPM / 1_000_000);
```

Comment:
For larger amounts it might be wise to first divide, and them multiply, in order to not overflow.

Reply:
Right, though I'm wondering about the protocol fee then, which does not have a minimum value,
which means, if we do the same thing there, we could end up with a protocol fee of zero for all amounts less than `1_000_000`. What's the best approach to handle this in Solidity?

Reply:
Good question, I have no answer. I guess in general (as in with the token that we'll be using) we don't expect users to transfer more than 2^256 - 10^6 token wei. SO I guess the current version is safe *enough*.